src/wx/opts.cpp

```diff
@@ -16,27 +16,27 @@
 /* not sure how well other compilers support field-init syntax */
 #define STROPT(c, n, d, v) \
     {                      \
-        wxT(c), (n), d, &v \
+        wxT(c), (n), d, &v, NULL, wxT(""), 0, 0, NULL, NULL, NULL, wxT(""), 0, 0, 0 \
```

Comment:
Would be better to set default values in the struct instead of this.

Reply:
The values are set I think. The issue is when initializing with missing fields. It gives this warning:
```
/home/denisfa/Git/visualboyadvance-m/src/wx/opts.cpp: At global scope:
/home/denisfa/Git/visualboyadvance-m/src/wx/opts.cpp:151:1: warning: missing initializer for member ‘wxJoyKeyBinding::mod’ [-Wmissing-field-initializers]
 };
 ^
/home/denisfa/Git/visualboyadvance-m/src/wx/opts.cpp:151:1: warning: missing initializer for member ‘wxJoyKeyBinding::joy’ [-Wmissing-field-initializers]
/home/denisfa/Git/visualboyadvance-m/src/wx/opts.cpp:151:1: warning: missing initializer for member ‘wxJoyKeyBinding::mod’ [-Wmissing-field-initializers]
/home/denisfa/Git/visualboyadvance-m/src/wx/opts.cpp:151:1: warning: missing initializer for member ‘wxJoyKeyBinding::joy’ [-Wmissing-field-initializers]
...
```

Reply:
I mean set default initializers for `struct opt_desc` in `opts.h`, see here:

https://stackoverflow.com/a/16783513/262458

e.g.

```cpp
struct foo {
  bool good = true;
  char* str = "something";
}
```

Reply:
This did not work. I had a lot of compilation issues when trying to initialize values from inside the struct. The link above has an example that did not help me on this code. I have an alternative, though. I can also expand on some of the issues, in case you feel we can get this working this way.

Reply:
It probably didn't work because all values except the first ones have to have initializers in order to not specify them, but otherwise it should work fine.